src/coreclr/vm/field.cpp

```diff
@@ -227,49 +227,74 @@ PTR_VOID FieldDesc::GetStaticAddressHandle(PTR_VOID base)
     }
     CONTRACTL_END
 
-    _ASSERTE(IsStatic());
+    LOG((LF_CORDB, LL_INFO1000, "FD::GSAH: this=%p, token=0x%08x, base=%p\n", this, GetMemberDef(), (void*)base));
+
+    PTR_VOID retVal = NULL;
+
 #ifdef FEATURE_METADATA_UPDATER
     if (IsEnCNew())
     {
-        EnCFieldDesc * pFD = dac_cast<PTR_EnCFieldDesc>(this);
-        _ASSERTE_IMPL(pFD->GetApproxEnclosingMethodTable()->SanityCheck());
-        _ASSERTE(pFD->GetModule()->IsEditAndContinueEnabled());
-
-        EditAndContinueModule *pModule = (EditAndContinueModule*)pFD->GetModule();
-        _ASSERTE(pModule->IsEditAndContinueEnabled());
-
-        PTR_VOID retVal = NULL;
+        LOG((LF_ENC, LL_INFO1000, "FD::GSAH: EnC field\n"));
 
 #ifdef DACCESS_COMPILE
         DacNotImpl();
 #else
+        EnCFieldDesc * pFD = dac_cast<PTR_EnCFieldDesc>(this);
+        _ASSERTE_IMPL(pFD->GetApproxEnclosingMethodTable()->SanityCheck());
+        _ASSERTE(pFD->GetModule()->IsEditAndContinueEnabled());
+
+        EditAndContinueModule *pEnCModule = (EditAndContinueModule*)pFD->GetModule();
+        if (IsRVA())
+        {
+            LOG((LF_ENC, LL_INFO1000, "FD::GSAH: EnC - RVA\n"));
+            retVal = (void*)pEnCModule->GetDynamicRvaField(pFD->GetMemberDef());
+        }
+        else
         {
             GCX_COOP();
             // This routine doesn't have a failure semantic - but Resolve*Field(...) does.
             // Something needs to be rethought here and I think it's E&C.
             CONTRACT_VIOLATION(ThrowsViolation|FaultViolation|GCViolation);   //B#25680 (Fix Enc violations)
-            retVal = (void *)(pModule->ResolveOrAllocateField(NULL, pFD));
+            retVal = (void*)(pEnCModule->ResolveOrAllocateField(NULL, pFD));
         }
 #endif // !DACCESS_COMPILE
+
+        LOG((LF_ENC, LL_INFO1000, "FD::GSAH: EnC - retVal=%p\n", (void*)retVal));
         return retVal;
     }
 #endif // FEATURE_METADATA_UPDATER
 
-    if (IsRVA())
-    {
-        Module* pModule = GetModule();
-        PTR_VOID ret = pModule->GetRvaField(GetOffset());
+    DWORD offset = GetOffset();
+    LOG((LF_CORDB, LL_INFO1000, "FD::GSAH: Offset=0x%x\n", offset));
 
-        _ASSERTE(pModule->IsReflectionEmit() || !pModule->IsRvaFieldTls(GetOffset()));
-
-        return(ret);
+    if (!IsRVA())
+    {
+        CONSISTENCY_CHECK(CheckPointer(base));
+        retVal = PTR_VOID(dac_cast<PTR_BYTE>(base) + offset);
     }
+    else
+    {
+        LOG((LF_CORDB, LL_INFO1000, "FD::GSAH: RVA field\n"));
 
-    CONSISTENCY_CHECK(CheckPointer(base));
-
-    PTR_VOID ret = PTR_VOID(dac_cast<PTR_BYTE>(base) + GetOffset());
+        Module* pModule = GetModule();
+        if (offset == FIELD_OFFSET_DYNAMIC_RVA)
```

Comment:
Why do we need this handled here? It should be covered by the `IsEnCNew` path above.